### ISSUE TITLE
Feature Request: Holiday Mode

### ISSUE THREAD

**Issue by ceee5** [MindrustUK/Heatmiser-for-home-assistant#142]
Hi,

In my quest to keep our home assistant deployment as transparent to the end users, I'd like to use the holiday mode, set either by the app or on our neoultra as a trigger to Home Assistant.

Can the Heatmiser for Home Assistant application support this as a standalone entity?

Many thanks 

Jon

**Comment by ceee5** [comment]
Great, thank you

**Comment by MindrustUK** [comment]
I've had a look at the API and this is indeed available as an option, and thanks to the great work already done by stikonas  it's already surfaced in the neohubapi library.

I'd need to now extend and surface the option as some sort of global setting across the Heatmiser devices, I'm thinking the Neohub as a central point is the way to go, creating entities and services related to that off the top of my head.

Can't promise any time for getting round to it but I'll put it on the list somewhere and try and revisit it once I clear out some of my back log.

**Comment by ocrease** [comment]
@ceee5 I think on the dev branch you should be able to do what you wanted now. It is not completely clear what you were requesting, but I think both of these are doable:

- If you want to trigger an automation when the heatmiser system is set to away or holiday, there are two new sensors (disabled by default) that show the away / holiday state:
![image](https://github.com/user-attachments/assets/184d87ec-0554-4668-960c-055ad8f011a3)
- If you want to be able to set away mode from HA, this is also now possible:
![image](https://github.com/user-attachments/assets/b5d8545f-12e3-4140-8614-b88e530f4b33)

Note that setting Away mode on any thermostat will set all of them to Away. If you want to just set one thermostat to standby/frost mode, just set the mode to Off (instead of heat on the screenshot)

You could also use the standard `Climate: Set preset mode` or `Climate: Set HVAC mode` actions if you want to change the state from an automation.

See #206 for small improvements coming

**Comment by ocrease** [comment]
#223 includes away, holiday and holiday end entities which I think satisfy the original request. Is there anything else we should add?

**Comment by ocrease** [comment]
@MindrustUK I have added a service that can be used to set away but more importantly a holiday (eg away with an end date).

![image](https://github.com/user-attachments/assets/54c208db-eed1-4f13-b7f8-530bcabbb1a9)


```
action: heatmiserneo.set_away_mode
data:
  away: true
  end: "2025-01-01 09:00:00"
target:
  entity_id: binary_sensor.neohub_192_168_80_10_away
```

It can be called in three ways:

- Away set to false. This will disable away and/or holiday mode
- Away set to true without an end date. This sets away mode, eg permanently away
- Away set to ture with an end date. This sets holiday mode, eg away with a specified return date

I don't think there is anything else needed for this ticket, do you?

The are a couple of potential changes to consider:
- Combine the away and holiday sensors. If the new sensor (lets call it away) is on/true but the holiday end is unset, then it is away.  If holiday end is set, then it is a holiday. I don't think there is value in having the separate sensors, if anything it will add confusion to anyone using them if they have to check both the away and the holiday sensor
- Remove the away and holiday sensors from the individual thermostats/timers. 

Any thoughts on those two changes? Both are easy to do, but will be harder to change once the next proper release is in general use.

Here are the hub sensors added in the past:

![image](https://github.com/user-attachments/assets/b2891c37-e6fa-47b4-907c-4fe53233c987)

**Comment by MindrustUK** [comment]
@ocrease I think the changes you suggest make a lot of sense. It's better to have one synthetic holiday mode than multiple away devices to my mind.

The only use case I can think of where this would be an exception, i.e. you would require multiple devices to be away would be in the case of a guest house /outbuilding or similar that shares the heating system from the house... I think @PhillyGilly has a scenario like this? In this case you maybe on holiday but guests may want to control heating in their building.

**Comment by ocrease** [comment]
@MindrustUK OK I will see what's involved with combining the two binary sensors. 

Regarding the scenario you mentioned, that's not really supported by Heatmiser right? If you put the hub into holiday/away mode, it should apply to all devices (I think technically it might be possible to have it only apply to some, but their docs discourage it). The way to handle the scenario you mention is to put individual devices into standby, though of course you can't specify and end date for standby. Having said that, one could create an automation to take the devices out of standby at a set date. Users of this integration are using Home Assistant after all :-)

**Comment by MindrustUK** [comment]
@ocrease My thoughts exactly, just wanted to sound it out loud for a sanity check.

**Comment by PhillyGilly** [comment]
When I was converting my gardening shed into habitable accommodation and "designing" the controls, I briefly considered running it on a separate neo-hub. In the end I decided to run it as part of the house with the noted limitation on holiday mode. The only problem I had with this is that the standby/holiday temperature (12?) is higher than my "off" profile temperature (5C). I think that I was able to put the house into holiday and then switch garden-studio back on. I will check later ( can't see now as I think I accidentally deleted one of my repeaters :-(  ). But in practice I'm not going to use holiday mode in the winter so either way it's not a problem.

**Comment by ocrease** [comment]
@MindrustUK PR #230 has the combined away/holiday sensor. I think once that is merged this feature request can be marked as done/closed?

@PhillyGilly the standby temperature is configurable, through the Heatmiser App or via the Frost Temperature configuration entity provided by this integration:

![image](https://github.com/user-attachments/assets/2c8a846f-3717-47b0-9efe-bb26080df6c5)

It is disabled by default, but you can enable it if you want to use it

**Comment by MindrustUK** [comment]
@ocrease closing complete. Beta 12 released with these fixes.
